great_expectations/expectations/expectation.py

```diff
@@ -1226,6 +1226,14 @@ def validate_(  # noqa: PLR0913
     @property
     def configuration(self) -> ExpectationConfiguration:
         kwargs = self.dict(exclude_defaults=True)
+
+        # Always add back domain-specific keys if there's a value associated with them (even if they are defaults)
+        for key in self.domain_keys:
+            if key not in kwargs and key in self.__fields__:
+                default = self.__fields__[key].default
+                if default:
+                    kwargs[key] = default
```

Comment:
Not my favorite piece of logic but if I have this custom Expectation:
```python
 class ExpectColumnAgesToBeLegalAdult(gxe.ExpectColumnValuesToBeBetween):
        column: str = "ages"
        min_value: int = 18
        description: str = "column values must be a legal adult age"
```

the current `configuration` property will remove all defaults, resulting in empty kwargs. Without kwargs in the config, renderers are unable to function properly. They need to know what column to associate the object with at a minimum so I've opted to use this logic to ensure we always have it when we need it.

Note that this issue is a very small edge case concerning simple custom expectations that override domains. If we want to encourage users to always provide a domain or we don't like the logic I've used here, let's forgo my changes